Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -49,6 +49,10 @@ public class ProgramBuilder {
         return contextAnalyzer.context
     }
 
+    public var activeContexts: [ProgramContext] {
```

Comment:
Is it enough to just expose the "active" context, i.e. the topmost one on the stack? The "inactive" contexts (below the topmost one) shouldn't really be required from what I can tell.

Reply:
I've been running some tests and we would need to walk the stack and check all contexts. We would need this for nested blocks. Here's an example:
```
async function x(){
  for(;;) {
   //.. instructions in for loop
  }
 }
```
In the for loop above it should be valid to splice a `Await` instruction since it's contained within the async function so the splicer would need to figure out is there's an 'active' asyncFunction context.

I've got some changes to fix this that I'll push in a bit.

Reply:
But in this example, wouldn't you have the following context stack:
```
[.function, .asyncFunction, .loop]   // TOP
[.function, .asyncFunction]
[.default]
```
Because [the ContextAnalyzer will copy the existing context into the loop context](https://github.com/googleprojectzero/fuzzilli/blob/4219480e371bd86a4eb5c2810e3ca37adc1a0a69/Sources/Fuzzilli/FuzzIL/Analyzer.swift#L176)?

Reply:
I think with this approach to contexts, we must only check the topmost ("active") context, since otherwise, the following example would break:

```
...
BeginFunctionDefinition
    BeginObjectDefinition
        X
...
```

In this case, at `X` you'd have the following context stack:
```
[.objectDefinition]
[.default, .function]
[.default]
```
And so it'd now be invalid to insert operations that require the `.default` context, even though that context is on the stack.

Reply:
Ah yes you're right